File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {
    
    public static String message(String logLine) {
        // split the string using the colon, trim and output the second half of message
        return logLine.split(":")[1].trim();
    }

    public static String logLevel(String logLine) {
        // split the string using colon, set string to lowercase, remove brackets
        String a = logLine.split(":")[0].toLowerCase();
        return a.replaceAll("[\\[\\]]", "");
    }

    public static String reformat(String logLine) {
        // use previous methods to reformat
        return message(logLine) + " (" + logLevel(logLine) + ")";
    }
}

File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
    std::string message(std::string line) {
        // return the message-------------------------------
        // locate beginning of message, add 2 to start at beggining of message
        int colon = line.find(":") + 2;
        // return substring starting at beginning of message
        std::string newString = line.substr(colon);
        return newString;
    }

    std::string log_level(std::string line) {
        // return the log level-----------------------------
        // find the brackets and adjust by 1, up and down, to exclude them from substring
        int openBracket = line.find("[") + 1;
        int closeBracket = line.find("]") - 1;
        return line.substr(openBracket, closeBracket);
    }

    std::string reformat(std::string line) {
        // return the reformatted message-------------------
        /* More work
        std::string mess = message(line);
        std::string log = log_level(line);
        return mess + " (" + log + ")";
        */
        // Less work vvv
        return message(line) + " (" + log_level(line) + ")";
    }
}
